### PR TITLE
[BEAM-1871] Remove Google API client dependency from sdks/core

### PR DESCRIPTION
**Pull request by vikkyrk**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
- Remove usage of google api client base64 encoding
- Copy google api client versions of BackOff, BackOffUtils and Sleeper classes into sdks/core.
- Use an adapters to convert from beam to google api client interfaces.

### CODE REVIEW

**Comment by vikkyrk** [comment]
R: @davorbonaci @lukecwik this is the best I could come up with. Haven't verified tests pass, but ready for a first pass of review.

**Comment by vikkyrk** [comment]
Retest this please

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/BackOff.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.util;
+
+import java.io.IOException;
+
+/**
+ * Back-off policy when retrying an operation.
+ *
+ * <p><b>Note</b>: This interface is copied from google api client library to avoid its dependency.
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/BackOff.java]: google api -> Google API

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/BackOffUtils.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ */
+
+package org.apache.beam.sdk.util;
+
+import java.io.IOException;
+
+/**
+ * Utilities for {@link BackOff}.
+ *
+ * <p><b>Note</b>: This is copied from google api client library to avoid its dependency.
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/BackOffUtils.java]: google api -> Google API

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/Sleeper.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * Sleeper interface to use for requesting the current thread to sleep as specified in
+ * {@link Thread#sleep(long)}.
+ *
+ * <p>
+ * The default implementation can be accessed at {@link #DEFAULT}. Primarily used for testing.
+ * </p>
+ *
+ * <p><b>Note</b>: This interface is copied from google api client library to avoid its dependency.
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/Sleeper.java]: google api -> Google API

File: sdks/java/core/src/test/java/org/apache/beam/sdk/testing/SystemNanoTimeSleeper.java [hunk 4]
```diff
@@ -17,10 +17,11 @@
  */
 package org.apache.beam.sdk.testing;
 
-import com.google.api.client.util.Sleeper;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.locks.LockSupport;
```
Comment (lukecwik) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/testing/SystemNanoTimeSleeper.java]: remove line separating imports

File: sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/util/BackOffAdapter.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.util;
+
+import java.io.IOException;
+
+/**
+ * An adapter for converting between Beam and Google api client representation of backoff utilities.
```
Comment (lukecwik) [inline sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/util/BackOffAdapter.java]: Beam and Google api client representation of backoff utilities. -> Apache Beam and Google API client representations of backoffs.

File: sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/util/FastNanoClockAndSleeper.java [hunk 6]
```diff
@@ -15,7 +15,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.beam.sdk.testing;
```
Comment (lukecwik) [inline sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/util/FastNanoClockAndSleeper.java]: We are working on moving stuff out of org.apache.beam.sdk.util 
Most of what is left is GCS related, @dhalperi do you have a PR that does this?

Comment (vikkyrk) [inline sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/util/FastNanoClockAndSleeper.java]: Will fix this in another PR.

**Comment by vikkyrk** [comment]
Addressed comments.

**Comment by lukecwik** [comment]
Merging

**Pull request closed without merging**